src/sentry/static/sentry/app/components/group/suggestedOwners/suggestedOwners.tsx

```diff
@@ -0,0 +1,194 @@
+import React from 'react';
+
+import {assignToUser, assignToActor} from 'app/actionCreators/group';
+import withApi from 'app/utils/withApi';
+import withOrganization from 'app/utils/withOrganization';
+import Access from 'app/components/acl/access';
+import {Organization, Group, Event, Actor, Commit, Project, User} from 'app/types';
+import {Client} from 'app/api';
+
+import {findMatchedRules, Rules} from './findMatchedRules';
+import {SuggestedAssignees} from './suggestedAssignees';
+import {OwnershipRules} from './ownershipRules';
+
+type OwnerList = React.ComponentProps<typeof SuggestedAssignees>['owners'];
+
+type Committer = {
+  author: User;
+  commits: Array<Commit>;
+};
+
+type Props = {
+  api: Client;
+  organization: Organization;
+  project: Project;
+  group: Group;
+  event: Event;
+};
+
+type State = {
+  rules: Rules;
+  owners: Array<Actor>;
+  committers: Array<Committer>;
+};
+
+class SuggestedOwners extends React.Component<Props, State> {
+  state: State = {
+    rules: null,
+    owners: [],
+    committers: [],
+  };
+
+  componentDidMount() {
+    this.fetchData(this.props.event);
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (this.props.event && prevProps.event) {
+      if (this.props.event.id !== prevProps.event.id) {
+        //two events, with different IDs
+        this.fetchData(this.props.event);
+      }
+      return;
+    }
+
+    if (this.props.event) {
+      //going from having no event to having an event
+      this.fetchData(this.props.event);
+    }
+  }
+
+  async fetchData(event: Event) {
```

Comment:
Should we make async component out of this while we are refactoring?

Reply:
Please why do you think that it is not a good idea? it's a request... it must be async

Reply:
I would suggest leaving it as there's some different fetching logic here and this needs to be refactored a bit so we're not making duplicate requests to `/committers/`.